### PR TITLE
DA Improve Function Arguments

### PR DESCRIPTION
**Pull request by Dala0**
Fixes messy way of accessing functions. '%@' used to have to be passed in all the time for no good reason.

Tested on a Mac Pro workstation.

@fredex42

### CODE REVIEW

**Comment by Dala0** [comment]
[PlutoHelperAgent needs to be able to open projects](https://trello.com/c/PbmQZlYi/2370-plutohelperagent-needs-to-be-able-to-open-projects)

File: PlutoHelperAgent/AppDelegate.m [hunk 1]
```diff
@@ -98,8 +98,8 @@ -    (void)getUrl:(NSAppleEventDescriptor *)event
         
         NSString *projectid = [parts objectAtIndex:2];
         
-        [ProjectLockerAndKeychainFunctions get_data_from_server:@"%@/api/project/" :@"/files" :projectid completionHandler:^void (NSURLResponse *response,NSDictionary *filesResult){
-            [ProjectLockerAndKeychainFunctions get_data_from_server:@"%@/api/storage/" :NULL :filesResult[@"files"][0][@"storage"] completionHandler:^(NSURLResponse *response, NSDictionary *storageResult) {
+        [ProjectLockerAndKeychainFunctions get_data_from_server:@"/api/project/" :@"/files" :projectid completionHandler:^void (NSURLResponse *response,NSDictionary *filesResult){
+            [ProjectLockerAndKeychainFunctions get_data_from_server:@"/api/storage/" :NULL :filesResult[@"files"][0][@"storage"] completionHandler:^(NSURLResponse *response, NSDictionary *storageResult) {
```
Comment (fredex42) [inline PlutoHelperAgent/AppDelegate.m:102]: is this safe (passing a NULL value for the selector here)?  Would a NULL value not cause `ProjectLockerAndKeychainFunctions.m` to crash at line 111?

File: PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m [hunk 2]
```diff
@@ -108,7 +108,7 @@ + (NSDictionary *) load_data_from_keychain {
 + (NSURLSessionTask *) communicate_with_server:(NSString*)url :(NSString*)method :(NSString*)type :(NSDictionary*)body :(BOOL)send_cookie :(BOOL)test_connection
```
Comment (fredex42) [inline PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m]: is `(BOOL)test_connection` used any more? If not, please remove it

File: PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m [hunk 3]
```diff
@@ -108,7 +108,7 @@ + (NSDictionary *) load_data_from_keychain {
...
                          completionHandler:(void (^) (NSURLResponse*, NSDictionary *))completionHandlerBlock {
   
-    NSString *URLToUse = [NSString stringWithFormat: url, [[NSUserDefaults standardUserDefaults] stringForKey:@"project_locker_url"]];
+    NSString *URLToUse = [NSString stringWithFormat: @"%@%@", [[NSUserDefaults standardUserDefaults] stringForKey:@"project_locker_url"], url];
```
Comment (fredex42) [inline PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m]: I think that there is a builtin method in NSString to combine filesystem and URL paths - https://developer.apple.com/documentation/foundation/nsstring/1417198-pathwithcomponents?language=objc which might be better here, avoiding any problems with missing or doubling / separators

**Review by fredex42 — CHANGES_REQUESTED** [review]
see inline

File: PlutoHelperAgent/AppDelegate.m (hunk 1 above)
Comment (Dala0) [reply PlutoHelperAgent/AppDelegate.m:102]: No. This is not a problem.

File: PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m (hunk 2 above)
Comment (Dala0) [reply PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m]: No. Done.

File: PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m (hunk 3 above)
Comment (Dala0) [reply PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m]: That is not what is being done here so it is not relevant.

File: PlutoHelperAgent/AppDelegate.m (hunk 1 above)
Comment (fredex42) [reply PlutoHelperAgent/AppDelegate.m:102]: ok. what happens in this eventuality?

**Review by fredex42 — APPROVED** [review]
👍

File: PlutoHelperAgent/AppDelegate.m (hunk 1 above)
Comment (Dala0) [reply PlutoHelperAgent/AppDelegate.m:102]: The string is correctly constructed.

The line you mentioned is irrelevant to the arguments of get_data_from_server. To see what happens, have a look at line 218.

**Pull request merged**